src/catalog/catalogGraphQLClient.ts

```diff
@@ -18,4 +19,26 @@ export class CatalogGraphQLClient extends AbstractGraphQLClient {
 
     return await this.post(request);
   }
+
+  public async findByQuery(
+    query: CatalogQuery,
+  ): Promise<GetProductsType | null> {
+    const queryStr: string = this.stringifyQuery(query);
+
+    try {
+      return await this.find(queryStr);
+    } catch (error: any) {
+      const exception: PublicApiClientException = new PublicApiClientException(
+        error?.message,
+        error?.response?.config,
+        error?.response?.status,
+      );
+
+      const { mustRetry } = await this.handleError(exception);
+      if (mustRetry) {
+        return await this.find(queryStr);
+      }
```

Comment:
```
const exception: PublicApiClientException = new PublicApiClientException(
        error?.message,
        error?.response?.config,
        error?.response?.status,
);
```
can you move this section in AbstractHttpClient.ts

```
const { mustRetry } = await this.handleError(exception);
if (mustRetry) {
    return await this.find(queryStr);
}
```
and move this section in AbstractGraphQlClient.ts

Reply:
I agree for the first part.

But not for the second part, I think it is refactored enough because the handleError is already in the Parent. And I want to let an extendable part in error handling. However in this case we replay the request and the method this.find does not exist in the parent.